lib/data/repositories/plot/implementation/MockPlotRepository.dart

```diff
@@ -0,0 +1,142 @@
+
+import 'dart:async';
+import 'package:farmsmart_flutter/data/model/EntityCollectionInterface.dart';
+import 'package:farmsmart_flutter/data/model/NewStageEntity.dart';
+import 'package:farmsmart_flutter/data/model/PlotEntity.dart';
+import 'package:farmsmart_flutter/data/model/PlotInfoEntity.dart';
+import 'package:farmsmart_flutter/data/model/ProfileEntity.dart';
+import 'package:farmsmart_flutter/data/model/crop_entity.dart';
+import 'package:farmsmart_flutter/data/model/mock/MockPlot.dart';
+import '../PlotRepositoryInterface.dart';
+
+
+final _plotBuilder = MockPlotEntity();
+
+class MockPlotRepository implements PlotRepositoryInterface {
+  final _plotStreamController =  StreamController<List<PlotEntity>>.broadcast(); 
+  List<PlotEntity> _plots = [];
+
+  var _observers = Map<String,StreamController<PlotEntity> >();   
+
+  @override
+  Future<PlotEntity> addPlot({ProfileEntity toProfile, PlotInfoEntity plotInfo, CropEntity crop}) {
+    final entity = _plotBuilder.build();
+    _plots.add(entity);
+    _update();
+    return Future.value(entity);
+  }
+
+  @override
+  Future<List<PlotEntity>> getCollection(EntityCollection<PlotEntity> collection) {
+    return collection.getEntities();
+  }
+
+  @override
+  Future<List<PlotEntity>> getFarm(ProfileEntity forProfile) {
+    return Future.value(_plots);
+  }
+
+  @override
+  Future<PlotEntity> getSingle(String uri) {
+    for (var i = 0; i < _plots.length; i++) {
+      if(_plots[i].id == uri){
+        return Future.value(_plots[i]);
+      }
+    }
+    return Future.error(Error());
```

Comment:
You can simplify this block with a `singleWhere` instead of iterating the array. `singleWhere` will take the first item in the array matching the condition that you set inside the `singleWhere` also you can catch the `orElse` param if doesn't find any, in this case I think that you need to return a null so the `orElse` I think that is not needed.
```suggestion
    final plotEntity = _plots.singleWhere((plotEntity) => plotEntity.id == uri, orElse: () => null);
    return plotEntity != null ? Future.value(plotEntity) : Future.error(Error());
```

Reply:
nice

Reply:
👍